starter-code/src/clue.js

```diff
@@ -1,94 +1,148 @@
+        age:          36,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+        occupation:   "Socialité"
+        }
+        
+    , mrMustard = {
+        first_name:   "Jack",
+        last_name:    "Mustard",
+        color:        "yellow",
+        description:  "He is a former football player who tries to get by on his former glory",
+        age:          62,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+        occupation:   "Retired Football player"
+        }
+        
 
-mrsPeacock
-first_name:   Eleanor
-last_name:    Peacock
-color:        blue
-description:  She is from a wealthy family and uses her status and money to earn popularity
-age:          36
-image:        https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg
-occupation:   Socialité
+];
 
-mrMustard
-first_name:   Jack
-last_name:    Mustard
-color:        yellow
-description:  He is a former football player who tries to get by on his former glory
-age:          62
-image:        https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg
-occupation:   Retired Football player
+// Rooms' Collection
+var roomsArray = [
+'Dinning Room',
+'Conservatory',
+'Kitchen',
+'Study',
+'Library',
+'Billiard Room',
+'Lounge',
+'Ballroom',
+'Hall',
+'Spa',
+'Living Room',
+'Observatory',
+'Theater',
+'Guest House',
+'Patio',
+];
 
-// Weapons
+// Weapons Collection
+var weaponsArray = [
+{ name : "rope", 
+  weight: 10
+},
+{ name : "knife",
+  weight: 8
+},
+{ name : "candlestick",
+  weight: 2
+},
+{ name : "dumbbell",
+  weight: 30
+},
+{ name : "poison",
+  weight: 2
+},
+{ name : "axe",
+  weight: 15
+},
+{ name : "bat",
+  weight: 13
+},
+{ name : "trophy",
+  weight: 25
+},
+{ name : "pistol",
+  weight: 20
+}
+];
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+var randomSelector = function(myArray) {
+    var keysArr = Object.keys(myArray);
+    var numKeys = myArray.length;
+    var rndIndx = Math.floor(Math.random() * numKeys);
+    return myArray[keysArr[rndIndx]];
+}
 
-// Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+function pickMistery () {
+    var misteryEnvelope = [
+        randomSelector(charactersArray),
+        randomSelector(weaponsArray),
+        randomSelector(roomsArray)
+    ];
+    return misteryEnvelope;
+}
 
-// Characters Collection
-var charactersArray = [];
+function revealmystery(misteryEnvelope) {
+    console.log(
```

Comment:
don't use the console.log when you want to return results. Either you create a variable that you call in the return or you return the result directly.
For Jasmine to validate your code, the sentence needs to be "exact" (yes, technology is dumb some times ^^)